File: EASY_TO_BUY_BACK_END/CHECKOUT_SERVICE/src/main/java/com/easytobuy/service/CheckOutService.java
/**
 * 
 */
package com.easytobuy.service;
import com.easytobuy.entity.Purchase;
import com.easytobuy.entity.PurchaseResponse;

/**
 * @author Satish Reddy
 *
 */
public interface CheckOutService {
	PurchaseResponse placeOrder(Purchase purchase);

}

File: EASY_TO_BUY_FRONT_END/easyToBuy_Angular/src/app/services/check-out.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Purchase } from '../entity/purchase';
import { OrderTracking } from '../entity/order-tracking.model';

@Injectable({
  providedIn: 'root'
})
export class CheckOutService {
private purchaseURL ="http://localhost:9096"
  constructor(private httpClient: HttpClient) {}
  placeOrder(purchase: Purchase):Observable<OrderTracking>{
    return this.httpClient.post<OrderTracking>(`${this.purchaseURL}/api/checkout/purchase`,purchase);
  }
}
